SU2_DEF/include/drivers/CDiscAdjDeformationDriver.hpp

```diff
@@ -0,0 +1,148 @@
+/*!
+ * \file CDiscAdjDeformationDriver.cpp
+ * \brief Headers of the main subroutines for driving the projection of sensitivities.
+ * \author T. Economon, H. Kline, R. Sanchez, A. Gastaldi, H. Patel
+ * \version 7.4.0 "Blackbird"
+ *
+ * SU2 Project Website: https://su2code.github.io
+ *
+ * The SU2 Project is maintained by the SU2 Foundation
+ * (http://su2foundation.org)
+ *
+ * Copyright 2012-2022, SU2 Contributors (cf. AUTHORS.md)
+ *
+ * SU2 is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * SU2 is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with SU2. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#define ENABLE_MAPS
+#include "../../../Common/include/CConfig.hpp"
+#undef ENABLE_MAPS
+
+#include "../../../Common/include/drivers/CDriverBase.hpp"
+#include "../../../Common/include/fem/fem_geometry_structure.hpp"
+#include "../../../Common/include/geometry/CGeometry.hpp"
+#include "../../../Common/include/grid_movement/CSurfaceMovement.hpp"
+#include "../../../Common/include/grid_movement/CVolumetricMovement.hpp"
+#include "../../../Common/include/parallelization/mpi_structure.hpp"
+#include "../../../SU2_CFD/include/numerics/CGradSmoothing.hpp"
+#include "../../../SU2_CFD/include/output/CBaselineOutput.hpp"
+#include "../../../SU2_CFD/include/output/COutput.hpp"
+#include "../../../SU2_CFD/include/solvers/CBaselineSolver.hpp"
+#include "../../../SU2_CFD/include/solvers/CGradientSmoothingSolver.hpp"
```

Comment:
Why do you need to include some many things here? Reduce to what you need and move to the cpp when possible.